app/models/solidus_stripe/gateway.rb

```diff
@@ -85,17 +85,15 @@ def capture(amount_in_cents, payment_intent_id, options = {})
     end
 
     # Authorizes and captures a certain amount on the provided payment source.
-    def purchase(amount_in_cents, _source, options = {})
-      currency = options.fetch(:currency)
-
+    # @todo add support for purchasing custom amounts
+    def purchase(_amount_in_cents, source, options = {})
       # Charge the Customer instead of the card:
-      payment_intent = request do
-        Stripe::PaymentIntent.create({
-          amount: to_stripe_amount(amount_in_cents, currency),
```

Comment:
The amount should stay there, this is the only use-case in which custom amounts are not an issue since we create the intent and charge the amount all in the same place.

Reply:
But how can we do that? I mean, now the amount is taken from the order.

Reply:
Oh, right, I was confused by the namespace, I get it now. This goes back to the encapsulation being a bit off but we can fix it in another iteration 👍